### PR TITLE
PLANNER-1173: Fixed indictment issues

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**
For Icons, if it overflows, you can scroll to see the rest (no scrollbar is shown) (we might be able to indicate this using a shadow). I added a colour scheme: -Hard -> red, -Medium -> yellow, -Soft -> gray, ZERO -> light green, +Soft -> green, +Medium (impossible currently) -> light blue, +Hard (impossible currently) -> purple (modifiable via CSS). The shades get more/less strong depending on how positive/negative the scores are, up to a limit, linearly.

### CODE REVIEW

File: employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/viewport/shiftroster/ShiftGridObject.java [hunk 1]
```diff
@@ -99,87 +108,143 @@ private void refresh() {
...
+            setAnimationDelay(50);
+        }
+    }
+
+    // NOTE: if amount == 0, then it returns start
+    //       if amount >= maxAmount, then it returns end
+    //       else it returns start + amount * ((end - start)/maxAmount)
+    private double mapToRange(double amount, double start, double end, double maxAmount) {
+        if (amount < 0) {
+            throw new IllegalArgumentException("amount (" + amount + ") must be non-negative.");
```
Comment (ge0ffrey) [inline employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/viewport/shiftroster/ShiftGridObject.java]: Iddeally, a sentence (as it ends with a ".") should start with a capital, so I 'd say "The amount ("...

Comment (ge0ffrey) [reply employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/viewport/shiftroster/ShiftGridObject.java]: not blocking the merge

File: employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/viewport/shiftroster/ShiftGridObject.java [hunk 2]
```diff
@@ -99,87 +108,143 @@ private void refresh() {
...
+        } else if (score.getHardScore() > 0) {
+            setAnimationDelay(mapToRange(score.getHardScore(), 85, 100, 10));
+        } else if (score.getMediumScore() < 0) {
+            setAnimationDelay(mapToRange(-score.getMediumScore(), 30, 15, 1));
+        } else if (score.getMediumScore() > 0) {
+            setAnimationDelay(mapToRange(score.getMediumScore(), 70, 85, 10));
+        } else if (score.getSoftScore() < 0) {
+            setAnimationDelay(mapToRange(-score.getSoftScore(), 50, 30, 100));
+        } else if (score.getSoftScore() > 0) {
+            setAnimationDelay(mapToRange(score.getSoftScore(), 50, 70, 30));
```
Comment (ge0ffrey) [inline employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/viewport/shiftroster/ShiftGridObject.java]: It seems weird that the animation delay differs based on the score. I would presume the color differs, but not the delay of an animation.
The function "mapToRange" is not self explanatory.

**Comment by ge0ffrey** [comment]
Merged, suggests if applicable can be handled in a separate PR

File: employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/viewport/shiftroster/ShiftGridObject.java (hunk 2 above)
Comment (Christopher-Chianelli) [reply employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/viewport/shiftroster/ShiftGridObject.java]: `animation-delay` is what controls color and allows it to smoothly transition between different shades. Basically, all shifts are in a paused CSS "animation" where it changes color. Setting the `animation-delay` allows us to control where the animation is. This allow a shift with `20 soft` to be more green than a shift with `10 soft`, or a shift with `-10 hard` to be more red than a shift with `-1 hard`. `mapToRange` is essentially a `lerp` (Linear interpolation) between `start` and and `end` by `value/maxValue`.

File: employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/viewport/shiftroster/ShiftGridObject.java (hunk 1 above)
Comment (Christopher-Chianelli) [reply employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/viewport/shiftroster/ShiftGridObject.java]: Will address this in a future PR

**Pull request merged**